internal/gengapic/auxiliary_test.go

```diff
@@ -0,0 +1,447 @@
+	"github.com/google/go-cmp/cmp"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+	"github.com/googleapis/gapic-generator-go/internal/txtdiff"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func TestWrapperExists(t *testing.T) {
+	testCases := []struct {
+		desc, wantErrStr string
+		existing         map[string]operationWrapper
+		target           operationWrapper
+		want             bool
+	}{
+		{
+			desc: "existing match",
+			existing: map[string]operationWrapper{
+				"FooOperation": {
+					name:         "FooOperation",
+					responseName: protoreflect.FullName("google.example.v1.Foo"),
+					metadataName: protoreflect.FullName("google.example.v1.FooMetadata"),
+				},
+			},
+			target: operationWrapper{
+				name:         "FooOperation",
+				responseName: protoreflect.FullName("google.example.v1.Foo"),
+				metadataName: protoreflect.FullName("google.example.v1.FooMetadata"),
+			},
+			want: true,
+		},
+		{
+			desc: "existing response mismatch",
+			existing: map[string]operationWrapper{
+				"FooOperation": {
+					name:         "FooOperation",
+					responseName: protoreflect.FullName("google.example.v1.Bar"),
+					metadataName: protoreflect.FullName("google.example.v1.FooMetadata"),
+				},
+			},
+			target: operationWrapper{
+				name:         "FooOperation",
+				responseName: protoreflect.FullName("google.example.v1.Foo"),
+				metadataName: protoreflect.FullName("google.example.v1.FooMetadata"),
+			},
+			want:       true,
+			wantErrStr: "mismatched response_types",
+		},
+		{
+			desc: "existing metadata mismatch",
+			existing: map[string]operationWrapper{
+				"FooOperation": {
+					name:         "FooOperation",
+					responseName: protoreflect.FullName("google.example.v1.Foo"),
+					metadataName: protoreflect.FullName("google.example.v1.BarMetadata"),
+				},
+			},
+			target: operationWrapper{
+				name:         "FooOperation",
+				responseName: protoreflect.FullName("google.example.v1.Foo"),
+				metadataName: protoreflect.FullName("google.example.v1.FooMetadata"),
+			},
+			want:       true,
+			wantErrStr: "mismatched metadata_types",
+		},
+		{
+			desc:     "doesn't exist",
+			existing: map[string]operationWrapper{},
+			target: operationWrapper{
+				name:         "FooOperation",
+				responseName: protoreflect.FullName("google.example.v1.Foo"),
+				metadataName: protoreflect.FullName("google.example.v1.FooMetadata"),
+			},
+			want: false,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			a := auxTypes{
+				opWrappers: tC.existing,
+			}
+			got, gotErr := a.wrapperExists(tC.target)
+			if tC.want != got {
+				t.Errorf("wrapperExists(%+v): expected %v, got %v, with existing: %+v", tC.target, tC.want, got, tC.existing)
```

Comment:
nit: unless this is already a pattern in this project can we switch this to be in the style of got want instead of want got.

Reply:
You bet! Done.